module_utils/common.py

```diff
@@ -181,25 +180,34 @@ def equal_values(v1, v2):
         return v1 == v2
 
 
-def equal_objects(d1, d2):
+def equal_objects(d1, d2, compare_common_fields_only=True):
```

Comment:
Also, we need to update our docs (`Task idempotency` section in `user_guide.md.j2`) to show how idempotency works right now. Let's describe there how we compare objects (especially, when some field are missing) and when we consider them equal and not.